app/src/androidTest/java/ch/epfl/sdp/blindly/fake_module/FakeUserHelperModule.kt

```diff
@@ -46,6 +48,15 @@ open class FakeUserHelperModule {
         Mockito.`when`(user.getSignInIntent()).thenReturn(fakeIntent)
 
         Mockito.`when`(user.isLoggedIn()).thenReturn(true)
+
+        val fastTaskCompletionSource = TaskCompletionSource<Void>().apply {
+            setResult(null)
+        }
+        val fastSuccessfulTask = fastTaskCompletionSource.task
+        //TODO This is injected (checked with a throw Exception)
+        // but at the time when the button logout is clicked, the task is still null... To fix
+        Mockito.`when`(user.logout(context)).thenReturn(fastSuccessfulTask)
```

Comment:
```suggestion
        Mockito.`when`(user.logout(any())).thenReturn(fastSuccessfulTask)
```

Reply:
What's happening here is that Mockito only return on the given argument (`context`) which is not the same here (Application context) as the one given in the Settings page (Settings context). So I think the best way here is to always return a successful task for any argument.

Reply:
This is pure genius, thank you! :D